model/llm/llm.go

```diff
@@ -274,6 +277,16 @@ func (m *Model) repairSourceCodeFile(ctx task.Context, codeRepairArguments *eval
 	return assessment, nil
 }
 
+// Cost returns the cost of the model.
+func (m *Model) Cost() (cost float64) {
+	return m.cost
+}
+
+// SetCost sets the cost of a model.
+func (m *Model) SetCost(cost float64) {
+	m.cost = cost
+}
```

Comment:
Would prefer to have this in the constructor so it cannot be changed afterwards. But if you need to adapt the constructor in thousands of places you can also introduce a new one I think: `NewModelWithCost`.